automation/clients/auth0.py

```diff
@@ -0,0 +1,74 @@
+import requests
+import tenacity
+
+from automation.config import Auth0Config
+from .secrets import SecretsClient
+
+##########
+# CLIENT #
+##########
+
+AUTH0_API_TOKEN_SECRET = "auth0_api_token"
+
+
+def is_unauthorized(exception):
+    return (
+        isinstance(exception, requests.exceptions.HTTPError)
+        and exception.response.status_code == requests.codes.unauthorized
+    )
+
+
+class Auth0Client:
+    def __init__(self, conf: Auth0Config):
+        self._base_url = "https://" + conf.domain
+        self._api_url = self._base_url + "/api/v2"
+        self._client_id = conf.client_id
+        self._client_secret = conf.client_secret
+        self._secrets_client = SecretsClient()
```

Comment:
I think we should pass this in to the constructor.

Reply:
got you. this does mean the signature for the auth0 client is `Auth0Client(auth0_conf, google_cloud_conf)` which feels... wrong :/

Reply:
I think the constructor should be `Auth0Client(auth0_conf, secrets_client)`